torchgeo/datasets/wms.py

```diff
@@ -0,0 +1,145 @@
+                 and returns a transformed version
+            crs: An optional string or integer code that is a
+                valid EPSG code (without the 'EPSG:')
+
+        """
+        super().__init__(transforms)
+        self._url = url
+        self._res = res
+        if crs is not None:
+            self._crs = CRS.from_epsg(crs)
+        self._wms = WebMapService(url)
+        self._format = self._wms.getOperationByName("GetMap").formatOptions[0]
+        self._layers = list(self._wms.contents)
+
+        if layer is not None and layer in self._layers:
+            self.layer(layer, crs)
+
+    def layer(self, layer: str, crs: Optional[Union[str, int]] = None) -> None:
+        """Set the layer to be fetched.
+
+        Args:
+            layer: A string with the name of the layer you want to fetch.
+            crs: An optional string or integer code that is a valid EPSG
+                code (without the 'EPSG:')
+        """
+        self._layer = self._wms[layer]
+        self._layer_name = layer
+        coords = self._wms[layer].boundingBox
+        self.index = Index(interleaved=False, properties=Property(dimension=3))
+        self.index.insert(
+            0,
+            (
+                float(coords[0]),
+                float(coords[2]),
+                float(coords[1]),
+                float(coords[3]),
+                0,
+                9.223372036854776e18,
+            ),
+        )
+        if crs is None:
+            i = 0
+            while self._crs is None:
+                crs_str = sorted(self._layer.crsOptions)[i].upper()
+                if "EPSG:" in crs_str:
+                    crs_str = crs_str[5:]
+                elif "CRS:84":
+                    crs_str = "4326"
+                try:
+                    self._crs = CRS.from_epsg(crs_str)
+                except CRSError:
+                    pass
+
+    def getlayer(self) -> ContentMetadata:
+        """Return the selected layer object."""
+        return self._layer
+
+    def layers(self) -> list[str]:
+        """Return a list of availiable layers."""
+        return self._layers
+
+    def __getitem__(self, query: BoundingBox) -> dict[str, Any]:
+        """Retrieve image/mask and metadata indexed by query.
+
+        Args:
+            query: (minx, maxx, miny, maxy, mint, maxt) coordinates to index
+
+        Returns:
+            sample of image/mask and metadata at that index
+
+        Raises:
+            IndexError: if query is not found in the index
+        """
+        img = self._wms.getmap(
+            layers=[self._layer_name],
+            srs="epsg:" + str(self.crs.to_epsg()),
+            bbox=(query.minx, query.miny, query.maxx, query.maxy),
+            # TODO fix size
+            size=(500, 500),
+            format=self._format,
+            transparent=True,
+        )
+        sample = {"crs": self.crs, "bbox": query}
+
+        transform = transforms.Compose([transforms.ToTensor()])
```

Comment:
We don't normally use torchvision transforms. What is the output from wms, a numpy array? Let's just convert that directly to PyTorch without relying on PIL

Reply:
It's a JPEG or PNG image can I convert that directly to a tensor?

Reply:
T.ToTensor() also automatically divides by 255 which we should leave up to the user to decide. You can convert the image to a tensor using `torch.from_numpy(np.array(Image.open(path).astype(np.float32)))`. You can break this up over multiple lines so it's not too verbose